Naloga35/src/Naloga35.jl

```diff
@@ -0,0 +1,206 @@
+diferencialnih enačb. Enačbo drugega reda d²θ/dt² + (g/l)sin(θ) = 0 prevede na
+sistem dveh enačb prvega reda.
+
+# Argumenti
+- `sprememba`: Vektor, v katerega se shrani izračunana sprememba stanja (odvod stanja).
+- `stanje`: Vektor trenutnega stanja nihala `[kot, hitrost]`.
+- `parametri`: Vektor s parametri `(gravitacija, dolzina)`.
+- `t`: Trenutni čas (v tej enačbi ni eksplicitno uporabljen).
+"""
+function nihalo!(sprememba, stanje, parametri, t)
+    kot = stanje[1]
+    hitrost = stanje[2]
+    gravitacija, dolzina = parametri
+
+    # d(kot)/dt = kotna hitrost
+    sprememba[1] = hitrost
+    # d(hitrost)/dt = -(g/l) * sin(kot)
+    sprememba[2] = -(gravitacija/dolzina) * sin(kot)
+end
+
+"""
+    resi_nihalo(zacetni_kot; ...)
+
+Reši diferencialno enačbo za matematično nihalo za dani začetni kot.
+
+# Argumenti
+- `zacetni_kot::Float64`: Začetni odmik nihala v radianih.
+
+# Neobvezni argumenti
+- `koncni_cas::Float64 = 3.0`: Končni čas simulacije v sekundah.
+- `gravitacija::Float64 = 9.81`: Težni pospešek v m/s².
+- `dolzina::Float64 = 1.0`: Dolžina vrvice nihala v metrih.
+- `zacetna_hitrost::Float64 = 0.0`: Začetna kotna hitrost v rad/s.
+- `korak_shranjevanja::Float64 = 0.001`: Časovni korak za shranjevanje rezultatov.
+
+# Vrne
+- `casi`: Vektor časovnih točk.
+- `koti`: Vektor kotnih odmikov v ustreznih časovnih točkah.
+"""
+function resi_nihalo(zacetni_kot::Float64;
+                     koncni_cas::Float64 = 3.0,
+                     gravitacija::Float64 = 9.81,
+                     dolzina::Float64 = 1.0,
+                     zacetna_hitrost::Float64 = 0.0,
+                     korak_shranjevanja::Float64 = 0.001)
+
+    # Združevanje parametrov in začetnih pogojev.
+    parametri = (gravitacija, dolzina)
+    zacetno_stanje = [zacetni_kot, zacetna_hitrost]
+    casovno_obmocje = (0.0, koncni_cas)
+
+    # Definiranje problema diferencialne enačbe.
+    problem = ODEProblem(nihalo!, zacetno_stanje, casovno_obmocje, parametri)
+    
+    # Reševanje problema z uporabo numerične metode DP5 (Dormand-Prince 5).
+    # `saveat` določa, v katerih časovnih točkah naj se shrani rešitev.
+    resitev = solve(problem, DP5(), saveat=korak_shranjevanja)
+
+    # Ekstrakcija časov in kotov iz rešitve.
+    casi = resitev.t
+    koti = resitev[1,:] # Prva komponenta rešitve so koti.
+
+    return casi, koti
+end
+
+"""
+    oceni_periodo(casi, koti)
+
+Oceni periodo nihanja na podlagi časov in kotov. Perioda je ocenjena kot
+štirikratnik časa, ki ga nihalo potrebuje, da prvič doseže ravnovesno lego
+(kot = 0).
+
+# Argumenti
+- `casi`: Vektor časovnih točk.
+- `koti`: Vektor kotnih odmikov.
+
+# Vrne
+- Ocenjeno periodo v sekundah. Če nihalo ne doseže ravnovesne lege, vrne -1.
+"""
+function oceni_periodo(casi, koti)
+    indeks = 1
+    # Iščemo prvi indeks, kjer kot postane nepozitiven.
+    # To ustreza prvi četrtini nihaja.
+    while indeks <= length(koti) && koti[indeks] > 0
```

Comment:
Natančnost periode je omejena s korakom metode DP5. Boljšo natančnost dobite lahko z uporabo metod za reševanje nelinearnih enačb (perioda $p$ je rešitev nelinearne enačbe $y(t_0) = y(t_0 + p)$ ).